flytekit/lazy_import/lazy_module.py

```diff
@@ -1,9 +1,19 @@
 import importlib.util
 import sys
+import types
 
 LAZY_MODULES = []
 
 
+class DummyModule(types.ModuleType):
```

Comment:
nit: can we name this LazyModule?

Reply:
done

Reply:
> do we have any way adding pandas v1 and v2 in our CI/CD matrix? Not sure if that's overkill, but there are probably many teams out there stuck on v1.

I feel it's a bit overkill. All the pandas API we use in flytekit works in both pandas v1 and v2. I'll add it upon any user runs into any issue.